disco/types/base.py

```diff
@@ -421,3 +421,77 @@ def attach(cls, it, data):
 
 class SlottedModel(Model):
     __slots__ = ['client']
+
+
+class BitsetMap(object):
+    @classmethod
+    def keys(cls):
+        for k, v in six.iteritems(cls.__dict__):
+            if k.isupper():
+                yield k
+
+
+class BitsetValue(object):
```

Comment:
If we wanted to be fancy we could probably make this a function which returns a class the closes around `map` instead of storing it in a slot, but I don't think its really a big deal.